util.go

```diff
@@ -150,3 +150,28 @@ type uint64Slice []uint64
 func (p uint64Slice) Len() int           { return len(p) }
 func (p uint64Slice) Less(i, j int) bool { return p[i] < p[j] }
 func (p uint64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+type timerWrapper struct {
+	t *time.Timer
+	C <-chan time.Time
+}
+
+func newTimer(timeout time.Duration) (t timerWrapper) {
+	if timeout > 0 {
+		t.t = time.NewTimer(timeout)
+		t.C = t.t.C
+	}
+	return
+}
+
+func (t *timerWrapper) Reset(d time.Duration) {
+	if t.t != nil {
+		t.t.Reset(d)
+	}
+}
```

Comment:
I think `Reset` will need to handle the case where `t == nil` and call `newTimer`

```go
t := newTimer(0)
t.Reset(1000)
```
Currently, calling `Reset` will not behave like `(*time.Timer).Reset` in this case